apps/explorer/src/app/hooks/use-txs-data.ts

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useFetch } from '@vegaprotocol/react-helpers';
+import type {
+  BlockExplorerTransactionResult,
+  BlockExplorerTransactions,
+} from '../routes/types/block-explorer-response';
+import { DATA_SOURCES } from '../config';
+
+export interface TxsStateProps {
+  txsData: BlockExplorerTransactionResult[];
+  hasMoreTxs: boolean;
+  lastCursor: string;
+}
+
+export interface IUseTxsData {
+  limit?: number;
+  filters?: string;
+}
+
+export const useTxsData = ({ limit = 10, filters }: IUseTxsData) => {
+  const [{ txsData, hasMoreTxs, lastCursor }, setTxsState] =
+    useState<TxsStateProps>({
+      txsData: [],
+      hasMoreTxs: true,
+      lastCursor: '',
+    });
+
+  let url = `${DATA_SOURCES.blockExplorerUrl}/transactions?limit=${limit}`;
+
+  if (filters) {
+    url = `${url}&${filters}`;
```

Comment:
suggestion (non-blocking): you can consider just putting this in a separate function, like `getUrl(base, limit, filters)`, instead of polluting the component with conditional variable assignments